spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitTemplatePublisherCallbacksIntegrationTests.java

```diff
@@ -205,26 +190,18 @@ public Void doInRabbit(Channel channel) throws Exception {
 	public void testPublisherConfirmWithSendAndReceive() throws Exception {
 		final CountDownLatch latch = new CountDownLatch(1);
 		final AtomicReference<CorrelationData> confirmCD = new AtomicReference<CorrelationData>();
-		templateWithConfirmsEnabled.setConfirmCallback(new ConfirmCallback() {
-
-			@Override
-			public void confirm(CorrelationData correlationData, boolean ack, String cause) {
-				confirmCD.set(correlationData);
-				latch.countDown();
-			}
+		templateWithConfirmsEnabled.setConfirmCallback((correlationData, ack, cause) -> {
+			confirmCD.set(correlationData);
+			latch.countDown();
 		});
 		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(this.connectionFactoryWithConfirmsEnabled);
 		container.setQueueNames(ROUTE);
-		container.setMessageListener(new MessageListenerAdapter(new Object() {
-
-			@SuppressWarnings("unused")
-			public String handleMessage(String in) {
-				return in.toUpperCase();
-			}
-		}));
+		container.setMessageListener(
+				new MessageListenerAdapter((ReplyingMessageListener<String, String>) in -> in.toUpperCase()));
```

Comment:
No, I mean that we have to mention it in the docs somewhere as a build-in variant to achieve a Lambda configuration.
Otherwise end-user may create its own variant of this interface with the same method to meet `MessageListenerAdapter` requirements. And once he will be surprised that we have one already but it isn't documented. 
This is end-user high level API without any usage in the Framework. So, should be mentioned in Docs. 
No? :smile: